### PR TITLE
Fix authentication to access dsg notebooks

### LINKED ISSUE #730: [Bug]: No authentication required to access DSG Notebooks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When a workbench is created with the DSG project, JupyterLab is launched without an authentication layer, and any one can access it.

### Expected Behavior

Every notebook to be protected by oauth proxy.

### Steps To Reproduce

1. In the DSG page, create a workbench and note its URL
2. In an incognito page or from any other computer, go to the notebook URL

### Workaround (if any)

Add the label to every notebook to authenticate: 

`oc annotate notebook/<notebook-name> notebooks.opendatahub.io/inject-oauth=true`

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

```yml
1.19.0
```

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Fixes #730 

## How Has This Been Tested?

1. Building a new image
2. Deploying it
3. Create a new notebook in dsg
4. Ensure is protected by Oauth Proxy.

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

File: frontend/src/api/network/notebooks.ts
Comment (bdattoma) [reply frontend/src/api/network/notebooks.ts:121]: I think it'd better to ask security team to jump in to understand if it can cause issues, what do you think?

Comment (andrewballantyne) [reply frontend/src/api/network/notebooks.ts:121]: This is k8s -- they can do whatever they want. I think you should make a post in the `eng` channel and tag Samu as well as me and if you know someone from Security. This is sorta by design -- although I can see the gap you're pointing out. It's an edge case and definitely worth a chat with security about it.

**Review by andrewballantyne — APPROVED** [review]
Tested the k8s routes on cluster with and without the applied oauth flag to the Notebook. Works as expected.

**Comment by andrewballantyne** [comment]
/hold

Have a question for Lucas... might be nothing.

**Pull request merged**
